src/compute/servers.rs

```diff
@@ -312,12 +311,42 @@ impl Server {
 
     /// Stop the server, optionally wait for it to be powered off.
     pub async fn stop(&mut self) -> Result<ServerStatusWaiter<'_>> {
-        api::server_simple_action(&self.session, &self.inner.id, "os-stop").await?;
+        self.action(ServerAction::Stop).await?;
         Ok(ServerStatusWaiter {
             server: self,
             target: protocol::ServerStatus::ShutOff,
         })
     }
+
+    /// Run an action on the server.
+    pub async fn action(&mut self, action: ServerAction) -> Result<()> {
+        api::server_action_with_args(&self.session, &self.inner.id, action).await
+    }
+}
+
+/// Serialize an enum unit variant into a None
+/// This is used to turn [ServerAction::Start] into
+/// `"os-start": null` instead of just `"os-start"`
+fn unit_to_null<S: serde::Serializer>(s: S) -> std::result::Result<S::Ok, S::Error> {
+    s.serialize_none()
+}
+
+/// An action to perform on a server.
+#[derive(Clone, Copy, Debug, Serialize)]
+pub enum ServerAction {
```

Comment:
Please mark as `#[non_exhaustive]` to prevent further additions from being breaking changes.